pemFioi/quickpi/blocklyQuickPi_lib.js

```diff
@@ -8826,10 +8832,65 @@ var getContext = function (display, infos, curLevel) {
         context.runner.waitDelay(callback, millis);
     };
 
-    context.quickpi.getTemperature = function(location, callback) {
-        var retVal =  25;
 
-        context.waitDelay(callback, retVal);
+    // TODO: change url to the "prod" one
+    context.quickpi.getTemperatureFromCloudUrl = "https://mapadev.com/nicolastest/weather.php";
+
+    // setup the supported towns
+    $.get(context.quickpi.getTemperatureFromCloudUrl + "?q=" + "supportedtowns", function(towns) {
+        context.quickpi.getTemperatureFromCloudSupportedTowns = towns;
+    });
+
+    // We create a cache so there is less calls to the api and we get the results of the temperature faster
+    context.quickpi.getTemperatureFromCloudCache = [];
+
+    context.quickpi.getTemperatureFromCloud = function(location, callback) {
+
+        function _updateGetTemperatureFromCloudCache(cache, location, newVal) {
```

Comment:
With the remark about making the cache an object (below), this function can probably be removed.

Reply:
Function removed on next push